analytics/lib/counts.py

```diff
@@ -188,23 +188,36 @@ def do_aggregate_to_summary_table(stat: CountStat, end_time: datetime,
             stat.property, (end - start) * 1000, cursor.rowcount))
 
     # Aggregate into InstallationCount
-    installationcount_query = """
-        INSERT INTO analytics_installationcount
-            (value, property, subgroup, end_time)
-        SELECT
-            sum(value), '%(property)s', analytics_realmcount.subgroup, %%(end_time)s
-        FROM analytics_realmcount
-        WHERE
-            property = '%(property)s' AND
-            end_time = %%(end_time)s
-        GROUP BY analytics_realmcount.subgroup
-    """ % {'property': stat.property}
+    filters = {'property': stat.property, 'end_time': end_time}
+    if realm is not None:
+        filters['realm_id'] = realm.id
+
+    realm_count = RealmCount.objects.filter(**filters)
+    total_value = {}  # type: Dict[str, int]
+    total_value = defaultdict(lambda: 0, total_value)
+
     start = time.time()
-    cursor.execute(installationcount_query, {'end_time': end_time})
+    for realm_count_ in realm_count:
+        subgroup = realm_count_.subgroup
+        total_value[subgroup] = total_value[subgroup] + realm_count_.value
+
+    row_count = 0
+    for subgroup, value in total_value.items():
+        filters = {'property': stat.property,
+                   'subgroup': subgroup, 'end_time': end_time}
+
+        if InstallationCount.objects.filter(**filters).exists():
+            installation_count = InstallationCount.objects.filter(**filters).first()
+            installation_count.value = installation_count.value + value
+            installation_count.save()
+        else:
+            filters['value'] = value
+            InstallationCount.objects.create(**filters)
+            row_count = row_count + 1
+
     end = time.time()
     logger.info("%s InstallationCount aggregation (%dms/%sr)" % (
-        stat.property, (end - start) * 1000, cursor.rowcount))
-    cursor.close()
```

Comment:
It seems like removing this line is a bug; we should be closing the cursor when we're done with it.